client/util/routes.ts

```diff
@@ -30,24 +30,44 @@ const getStatusUrl = (category: string, children: string): string => {
     return getJobUrl(category, children, "/status");
 };
 
+const getPidActionUrl = (pid: string, action: string): string => {
+    return `${editObjectUrl}/${encodeURIComponent(pid)}/${action}`;
+}
+
+const getObjectChildrenUrl = (pid: string): string => {
+    return pid.length > 0 ? getPidActionUrl(pid, "children") : `${apiUrl}/edit/treeTop`;
+}
+
+const getObjectDetailsUrl = (pid: string): string => {
+    return getPidActionUrl(pid, "details");
+}
+
 const getObjectModelsDatastreamsUrl = (pid: string): string => {
-    return `${editObjectUrl}/modelsdatastreams/${pid}`;
+    return getPidActionUrl(pid, "modelsdatastreams");
+}
+
+const getObjectParentsUrl = (pid: string): string => {
+    return getPidActionUrl(pid, "parents");
+}
+
+const getDatastreamActionUrl = (pid: string, datastream: string, action = ""): string => {
+    return getPidActionUrl(pid, `datastream/${encodeURIComponent(datastream)}` + (action.length > 0 ? `/${action}` : ""));
```

Comment:
if the string is empty, wouldn't this still call the same url regardless if the backslash exists or not?

Reply:
Yes, that's true. We could probably simplify this a little bit, though it just feels "cleaner" to me to omit the slash when we don't need it.

Reply:
ok